js-core/week-05/homework.md

```diff
@@ -1,44 +1,101 @@
-
-When reading always try the code snippets by yourself in Node. (change
-document.writeln to console.log)
-
-Next class - we want you to do a presentation (including showing some code
-sample) for one of these concepts:
-
-* Switch Statements and Ternary Operators
-* Math methods in JavaScript
-* While loops (compare them to for Loops)
-* Array.slice and Array.splice
-* Array.map and Array.forEach
-* Array.find and Array.filter
-* String.replace, String.search and other String methods
-* Object.keys (How to loop the properties of an object)
-* [Function arguments](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Functions/arguments)
-* [Advanced][different ways of creating objects](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Working_with_Objects#Creating_new_objects)
-* [Advanced][inheritance](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Working_with_Objects#Inheritance)
-* [Advanced][regular expressions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions)
-
-## Prepare for the next class
-
-1. Reserach **Unit Testing** (next time, know how to answer the question: **what
-   is TDD?**)
+
+### Challenges
+Go back to [the repo](https://github.com/Code-Your-Future/JS-Core-1-Exercises)
+that you have forked during the last class:
+* If you haven't attempted all the week 1 base challenges, go back and work on
+them!
+* If you're all done, challenge yourself with the week 1 advanced challenges,
+and the ones present in week 2.
+
+
+### Code style and best practices
+As you might have noticed, there are often several _valid_ ways to write the
+same code.
+
+For example we can define a string with either single or double:
+quotes:
+```js
+var singleQuotes = 'Hello';
+var doubleQuotes = "Hello";  // exactly the same as singleQuotes
+
+// check if they're actually the same
+singleQuotes == doubleQuotes  // true
+```
+
+Another example, on spacing and newlines:
+```js
+function sayHello() {
+  return 'Hello world!';
+}
+
+// is the same as
+function sayHello (){
+  return 'Hello world!';
+}
+
+// or
+function sayHello()
+{
+  return 'Hello world!';
+}
+
+// and even
+function sayHello(){return 'Hello world!';}
+```
+
+While functionally the same, good developer are **consistent** in the way they
+write their code.
+
+Consistency and predictability in the _visual structure_ of the code won't
+change the way the computer will execute the code, but will make it way easier
+to maintain the code and spot bugs for both you and other programmers.
+
+To this end, many organisation and people develop a set of _conventions_ ,
+or **style guide** on how to structure their code. While often one option is not strictly 'better' than others, it is tremendously helpful if all the people
+working on a project write in the same style.
+
+To get some more insight on the argument, read these links:
+* https://www.smashingmagazine.com/2012/10/why-coding-style-matters/ dig a bit
```

Comment:
+1 for content though it's nicer to not drop the full link in the text but use its title and link